src/openrct2/rct1/S4Importer.cpp

```diff
@@ -2945,8 +2945,11 @@ class S4Importer final : public IParkImporter
     {
         const auto originalString = _s4.string_table[(stringId - USER_STRING_START) % 1024];
         std::string_view originalStringView(originalString, USER_STRING_MAX_LENGTH);
-        auto withoutFormatCodes = RCT12::RemoveFormatCodes(originalStringView);
-        return rct2_to_utf8(withoutFormatCodes, RCT2_LANGUAGE_ID_ENGLISH_UK);
+        auto asUtf8 = rct2_to_utf8(originalStringView, RCT2_LANGUAGE_ID_ENGLISH_UK);
+        auto buf = std::make_unique<char[]>(asUtf8.size() + 1);
+        std::strcpy(buf.get(), asUtf8.data());
+        utf8_remove_format_codes(buf.get(), /*allow colour*/ false);
+        return buf.get();
```

Comment:
```suggestion
        auto asUtf8 = rct2_to_utf8(originalStringView, RCT2_LANGUAGE_ID_ENGLISH_UK);
        std::string buf(asUtf8.data(), asUtf8.size() + 1);
        utf8_remove_format_codes(buf.data(), /*allow colour*/ false);
        return buf;
```

Reply:
I avoided that because utf8_remove_format_codes will insert a "\0" in the middle of the string. So this string will have a `size()` that is different from `strlen(str.c_str())`. I wasn't sure if the code receiving this string is prepared to handle that. Sometimes developers will assume that.

Reply:
I'm aware. But in this context we use it as string anyway and don't change the actual buffer length. It still gets converted to string in the end, so the buffer would be truncated at that point. You can add a note to let this be known.

Reply:
Fair enough. I'll change that.

Reply:
I actually had to return `buf.data()` instead of `buf` to strip the remaining bytes after the '\0' in the string. The test was still broken if I don't do that. :(